examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +127,38 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    pid_t cpid;
+    int wstatus;
+    int fd;
+
+    // create a file with 644 permission
+    fd = open(outputfile, (O_WRONLY|O_TRUNC|O_CREAT),
+                (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH));
+    if (fd < 0)
+        return false;
+
+    cpid = fork();
+    if (cpid == -1) {
+        close(fd);
+
+        return false;
+    }
+
+    if (cpid == 0) {    /* child */
+        dup2(fd, 1);
+        close(fd);
+        if (execv(command[0], command) == -1)
+            exit(EXIT_FAILURE);
+    } else {            /* parent */
+        close(fd);
+        do {
+            if (waitpid(cpid, &wstatus, 0) == -1)
+                return false;
+            if (WEXITSTATUS(wstatus) != 0)
+                return false;
```

Comment:
Nice work! This implementation should work to avoid missing cleanup on va_end.  However, see suggestions in class regarding avoiding duplication of logic with the function above.